database/src/main/scala/cromwell/database/slick/CallCachingSlickDatabase.scala

```diff
@@ -37,20 +37,12 @@ trait CallCachingSlickDatabase extends CallCachingStore {
     }
   }
 
-  override def fetchCachedResult(metaInfoId: MetaInfoId)(implicit ec: ExecutionContext): Future[Option[CachedResult]] = {
+  override def fetchCachedResult(metaInfoId: MetaInfoId)(implicit ec: ExecutionContext): Future[CachedResult] = {
     val action = for {
-      metaInfo <- dataAccess.metaInfoById(metaInfoId.id).result
+      metaInfo <- dataAccess.metaInfoById(metaInfoId.id).result.head
```

Comment:
@mcovarr it does mean however that now every single code change needs to be more carefully inspected, and those code changes might end up in a diff such that it's not obvious it'll blow up.